gym_roboy/envs/msj_env.py

```diff
@@ -63,27 +65,24 @@ def render(self, mode='human'):
         pass
 
     @classmethod
-    def compute_reward(cls, achieved_goal, desired_goal, info):
-        assert len(achieved_goal) == MsjRobotState.DIM_JOINT_ANGLE, str(len(achieved_goal))
-        assert len(desired_goal) == MsjRobotState.DIM_JOINT_ANGLE, str(len(desired_goal))
-        current_joint_angle = achieved_goal
-        reward = -_l2_distance(current_joint_angle, desired_goal)
-        assert cls.reward_range[0] <= reward <= cls.reward_range[1]
-        return reward
+    def compute_reward(cls, current_state: MsjRobotState, goal_state: MsjRobotState, info=None):
+        reward = -_l2_distance(current_state.joint_angle, goal_state.joint_angle)
+        joint_vel_penalty = _l2_distance(current_state.joint_vel, goal_state.joint_vel)
+        total_reward = reward - joint_vel_penalty
```

Comment:
It would discouraging from moving fast generally yes. But staying away from the target only accumulates negative reward. Hard to tell what the general effect is.